mautic/mautic
# Bump php-cs-fixer from 2.16.10 to 2.19.2

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | N/A
| New feature/enhancement? (use the a.x branch)      | N/A
| Deprecations?                          | no
| BC breaks? (use the c.x branch)        | no
| Automated tests included?              | N/A
| Related user documentation PR URL      | N/A
| Related developer documentation PR URL | N/A
| Issue(s) addressed                     | Fixes #... 



#### Description:

This bump is needed for two reasons:
- Marketplace installing/removing PR: https://github.com/mautic/mautic/pull/10518 - the `composer/composer` package we're adding there requires `composer/xdebug-handler` 2.0 and higher - php-cs-fixer was blocking this, but they added support in [2.18.6](https://packagist.org/packages/friendsofphp/php-cs-fixer#v2.18.6)
- PHP 8 support ([added in 2.17.0](https://packagist.org/packages/friendsofphp/php-cs-fixer#v2.17.0)) - needed for https://github.com/mautic/mautic/pull/9969

Some remarks:
- `.php_cs` was deprecated. Renamed to `.php-cs-fixer.php`
- `PhpCsFixer\Config::create()` was deprecated. Replaced with `new PhpCsFixer\Config()`
- `binary_operator_spaces` config items were deprecated. Updated in https://github.com/mautic/mautic/pull/10519/commits/50584fddb1f140d2345641fa90a8d8418ae19da9
- Disabled `no_alternative_syntax` as it replaces all `endif`, `endforeach`, etc. with curly brackets, which we use extensively in our PHP templates. It led to 800+ changed files. We probably want to wait with this until we've switched to Twig for templating (required in Symfony 5)



#### Steps to test this PR:


1. Load up [this PR](https://mautibox.com)
2. Check if Mautic's functionality still works as expected. No functional changes should be observed, but this is largely covered by automated tests already.
